app/views/api.py

```diff
@@ -527,3 +533,25 @@ def update(self, request, *args, **kwargs):
             leave.hr_comment,
         )
         return response
+
+
+class VerifyEmailAPIView(GenericAPIView):
+    serializer_class = serializers.VerifyEmailSerializer
+    permission_classes = (AllowAny,)
+
+    def post(self, request):
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        email = serializer.validated_data["email"]
+        if not models.User.objects.filter(email=email).exists():
```

Comment:
All of this can be done in the `validate` function of the said serializer.
You won't have to return any `Response`.